### PR TITLE
ROC-2819: Create claim using API call in majority of tests

### PR DESCRIPTION
**Pull request by geek-soft**
### JIRA link ###

https://tools.hmcts.net/jira/browse/ROC-2819

### Change description ###

This PR creates claims using API call as opposite to clicking through UI.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/helpers/claimStoreHelper.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { ClaimStoreClient } from 'helpers/clients/claimStoreClient'
+import { IdamClient } from 'helpers/clients/idamClient'
+
+class ClaimStoreHelper extends codecept_helper {
+
+  async createClaim (claimData: ClaimData, userEmail: string): Promise<string> {
```
Comment (geek-soft) [inline src/helpers/claimStoreHelper.ts]: @pragnesh-patel, is `createClaim` fine or you prefer other wording like `makeClaim` for example?

**Comment by rlewan** [comment]
❤️ 
@geek-soft what's the time gain?

**Comment by geek-soft** [comment]
@rlewan, I saw built to finish within 10 minutes at night (ideal conditions). Master one finishes in around 16 minutes. We need to remember that runs all tests including T2 ones, so T1 exec time will be shorter.

File: src/tests/citizen/ccj/pages/ccj-check-and-send.ts [hunk 2]
```diff
@@ -27,34 +27,34 @@ export class CountyCourtJudgementCheckAndSendPage {
     I.click(buttons.submit)
   }
 
-  checkDefendantName (defendant, defendantType: PartyType): void {
+  checkDefendantName (defendant: Party, defendantType: PartyType): void {
```
Comment (geek-soft) [inline src/tests/citizen/ccj/pages/ccj-check-and-send.ts]: In bunch similar methods it makes sense to only pass the defendant object which has the type property. I plan to do this refactoring as separate PR to not make this PR even bigger.

File: types/model.d.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
+declare type User = {
+  id: string
+  bearerToken: string
+}
+
+declare type ClaimData = {
```
Comment (geek-soft) [inline types/model.d.ts]: Quick safety net before we reuse our domain classes in integration tests. When that happened most of the type definitions will be removed from here.

File: types/steps-override.d.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+declare namespace CodeceptJS {
```
Comment (geek-soft) [inline types/steps-override.d.ts]: Moved our overrides to separate file so we can more easily regenerate main steps definitions.

**Comment by geek-soft** [comment]
As I said earlier all tests (T1 + T2) run new in around 11 minutes, down by 4-5 minutes from initial 16 minutes. T1 tests run in around 7 minutes now.

The next step will be:
- reuse types from cmc-domain-model NPM module
- create response with API calls in offer tests

File: package.json [hunk 5]
```diff
@@ -12,7 +12,8 @@
   "scripts": {
     "lint": "tslint --project tsconfig.json && standard",
     "test": "ts-node --require tsconfig-paths/register ./node_modules/.bin/codeceptjs run --config default.conf.js --reporter mocha-multi",
-    "test-crossbrowser": "ts-node --require tsconfig-paths/register ./node_modules/.bin/codeceptjs run --config saucelabs.conf.js --grep @quick --reporter mocha-multi"
+    "test-crossbrowser": "ts-node --require tsconfig-paths/register ./node_modules/.bin/codeceptjs run --config saucelabs.conf.js --grep @quick --reporter mocha-multi",
+    "precommit": "yarn lint"
```
Comment (rlewan) [inline package.json]: That will make Pragnesh happy.

File: src/data/test-data.ts [hunk 6]
```diff
@@ -1,10 +1,58 @@
...
+      state: {
+        status: 'success',
+        finished: true
+      },
+      amount: claimFee * 100,
+      reference: 'CMC1$$$4d308250-d89e-485b-aafb-33a641fd00b3$$$AA00$$$X0024',
+      description: 'Money Claim issue fee',
+      date_created: '1511175701404'
+    },
+    feeAmountInPennies: claimFee * 100,
```
Comment (rlewan) [inline src/data/test-data.ts]: Multiplication by 100 in 2 places -- extract a function?

File: src/tests/citizen/claim/pages/claimant-check-and-send.ts [hunk 7]
```diff
@@ -92,15 +94,17 @@ export class ClaimantCheckAndSendPage {
   }
 
   verifyClaimAmount (): void {
-    I.see('£' + claimant().claimAmount.getClaimTotal().toFixed(2))
-    I.see('£' + claimant().claimAmount.claimFee.toFixed(2))
-    I.see('£' + claimant().claimAmount.getTotal().toFixed(2))
+    I.see('£' + claimAmount.getClaimTotal().toFixed(2))
+    I.see('£' + claimFee.toFixed(2))
+    I.see('£' + claimAmount.getTotal().toFixed(2))
```
Comment (rlewan) [inline src/tests/citizen/claim/pages/claimant-check-and-send.ts]: Probably not a part of this PR but would be nice to have a `formatPounds` method.
